### ISSUE TITLE
[Vertigo] Bad ES replacement with OR clause

### ISSUE THREAD

**Issue by durandx** [KleeGroup/vertigo#46]
The OR clause in ES search is not correctly handed

Considering the following expression :
```
+(NOM_NAISSANCE:#+nomNaissance# OR NOM:#+nomNaissance#) +PRENOM:#+prenom# +DATE_MODIFICATION_DEPUIS:[#dateModificationDepuis#!(*) TO *] +DATE_NAISSANCE:#dateNaissance#!(*)
```

The expression generated by Vertigo is:
```
+( OR   +DATE_MODIFICATION_DEPUIS:["2998-12-30T23:00:00.000Z" TO *] +DATE_NAISSANCE:*)]
```
Resulting in a Parse Exception.
Parse Failure [Failed to parse source [{"from":0,"size":50,"query":{"query_string":{"query":" +( OR   +DATE_MODIFICATION_DEPUIS:[\"2998-12-30T23:00:00.000Z\" TO *] +DATE_NAISSANCE:*)",

As a workaround I've deleted this useless 'OR' clause, but it could be nice to have an assertion that prevent us to send Bad request to ES (maybe you can simply prohibit the usage of OR and AND clause).

**Comment by NPi2Loup** [comment]
Not simple to fix. We don't want to managed a advanced grammar analyzer.
For now OR and AND aren't mandatory and should not be use.

**Comment by NPi2Loup** [comment]
I can't reproduce in 0.9.0.
Which version and implementation of ListFilterBuilder you use ?

**Comment by durandx** [comment]
At the time of the issue, it was on 0.8.3

**Comment by NPi2Loup** [comment]
Fixed in 0.9.0
